File: ProyectoFinal/src/main/kotlin/com/proyecto/models/Evento.kt
package com.proyecto.models

import com.proyecto.serializer.LocalDateTimeSerializer
import com.proyecto.serializer.ObjectIdSerializer
import kotlinx.serialization.Serializable
import org.bson.BsonObjectId
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.types.ObjectId
import java.time.LocalDateTime

@Serializable
data class Evento(
    @BsonId
    @Serializable(with = ObjectIdSerializer::class)
    val id: ObjectId,
    var nombre: String,
    var descripcion: String,
    @Serializable(with =LocalDateTimeSerializer::class)
    var fecha: LocalDateTime,
    var lugar: String,
    var imagen: String?,
    var desafios: List<Desafio>,
    var ranking: List<Ranking>
)

File: my-app/app/home/shared/evento.ts
export interface Evento {
  desafios: Desafio[];
  ranking: Ranking[];
  descripcion: string;
  fecha: string;
  id: string;
  imagen: string;
  lugar: string;
  nombre: string;
}
export interface Ranking {
  userName: string;
  tiempo: string;
}

export interface Desafio {
  descripcion: string;
  direccion: Direccion;
  nombreDesafio: string;
  numDesafio: number;
  resultado: string;
  clave: string;
}

export interface Direccion {
  latitud: string;
  longitud: string;
}
